capa/features/extractors/ida/insn.py

```diff
@@ -81,11 +83,27 @@ def extract_insn_api_features(fh: FunctionHandle, bbh: BBHandle, ih: InsnHandle)
         # tuple (<module>, <function>, <ordinal>)
         for name in capa.features.extractors.helpers.generate_symbols(api[0], api[1]):
             yield API(name), ih.address
+        return
```

Comment:
imports are also calls to operands of type `o_mem`, which means the added code will extract it again. A `call` instruction should call only one function, so I think we can stop if a call to an import/extern is extracted. Let me know if there are any concerns with it.

Reply:
it doesn't make sense to me that `check_for_api_call` is an iterator since it can only ever generate one item (or zero). like you said. i think we should change that route to have the signature `def check_for_api_call(...) -> Optional[Tuple[dll, name]]`. would you like to make that change here?

i concur that the new `return` makes sense, but i think this is the more complete fix.

Reply:
ah, right, thanks @mike-hunhoff